presto-main/src/main/java/com/facebook/presto/execution/buffer/AESSpillCipher.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.buffer;
+
+import com.facebook.presto.spi.PrestoException;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+final class AESSpillCipher
+        implements SpillCipher
+{
+    public enum Mode
+    {
+        AES_128(128),
```

Comment:
128-bit keys seem to not be used in production. What's the reason for including them?

Reply:
This is there for a couple reasons, none of them great:
- AES 128 can be (just barely) faster than AES 256 and theoretically someone might want to opt into that behavior (although I did not include a config switch for that in this PR)
- The same `AesSpillCipher` implementation works for both, but 256-bit AES isn't universally available. Oracle JDK releases before 8u161 don't support 256-bit AES out of the box and required you to manually install the "Unlimited Strength Jurisdiction Policy" for each version installed, whereas 128-bit AES was supported out of the box. I figured at least some new developers checking out presto and running tests locally might bump into that issue.

Let me know if you'd like to see it behind a different config flag, removed, or fine with it as-is.

Reply:
@pettyjamesm James, thanks for explaining. For simplicity sake, let's remove this option. I suggest we explain any gotchas you know of in the docs. BTW, let's update Spill to Disk documentation to describe this new functionality: https://prestodb.github.io/docs/current/admin/spill.html

Reply:
This is removed in the latest revision.